crates/rome_js_analyze/src/suppression_action.rs

```diff
@@ -0,0 +1,220 @@
+                        (
+                            TriviaPieceKind::SingleLineComment,
+                            format!("// {}: suppressed ", suppression_text).as_str(),
+                        ),
+                        (TriviaPieceKind::Newline, "\n"),
+                    ])
+                };
+                mutation.replace_token_transfer_trivia(first_token_with_newline, new_token);
+            }
+        } else {
+            let mut new_token = first_token_with_newline.clone();
+            if !should_insert_leading_newline {
+                new_token = new_token.with_leading_trivia([
+                    (TriviaPieceKind::Newline, "\n"),
+                    (
+                        TriviaPieceKind::SingleLineComment,
+                        format!("// {}: suppressed ", suppression_text).as_str(),
+                    ),
+                    (TriviaPieceKind::Newline, "\n"),
+                ])
+            } else {
+                new_token = new_token.with_leading_trivia([
+                    (
+                        TriviaPieceKind::SingleLineComment,
+                        format!("// {}: suppressed ", suppression_text).as_str(),
+                    ),
+                    (TriviaPieceKind::Newline, "\n"),
+                ])
+            };
+            mutation.replace_token_transfer_trivia(first_token_with_newline, new_token);
+        }
+    }
+}
+
+/// It checks if the current token has leading trivia newline. If not, it
+/// it peeks the previous token and recursively call itself.
+///
+/// Due to the nature of JSX, sometimes the current token might contain text that contains
+/// some newline. In case that happens, we choose that token.
+///
+/// Due to the nature of JavaScript templates, we also check if the tokens we browse are
+/// `${` and if so, we stop there.
+fn find_first_token_with_newline(token: JsSyntaxToken) -> Option<(JsSyntaxToken, bool)> {
+    let mut current_token = token;
+    let mut should_insert_leading_newline = false;
+    loop {
+        let trivia = current_token.leading_trivia();
+        if trivia.pieces().any(|trivia| trivia.is_newline())
+            || current_token.text_trimmed().contains('\n')
+            || current_token.text_trimmed().contains("\r\n")
+        {
+            break;
+        } else if matches!(current_token.kind(), JsSyntaxKind::DOLLAR_CURLY)
+            && current_token
+                .parent()
+                .map(|p| JsTemplateElement::can_cast(p.kind()))
+                .unwrap_or_default()
+        {
+            if let Some(next_token) = current_token.next_token() {
+                current_token = next_token;
+                break;
+            } else {
+                continue;
+            }
+        } else if let Some(token) = current_token.prev_token() {
+            current_token = token;
+            continue;
+        } else {
+            should_insert_leading_newline = true;
+            break;
+        }
+    }
```

Comment:
Nit:

There are a few unnecessary `continue` calls and explicitly returning a value from the loop can improve readabilty

```suggestion
    let should_insert_leading_newline = loop {
        let trivia = current_token.leading_trivia();
        if trivia.pieces().any(|trivia| trivia.is_newline())
            || current_token.text_trimmed().contains('\n')
            || current_token.text_trimmed().contains("\r\n")
        {
            break false;
        } else if matches!(current_token.kind(), JsSyntaxKind::DOLLAR_CURLY)
            && current_token
                .parent()
                .map(|p| JsTemplateElement::can_cast(p.kind()))
                .unwrap_or_default()
        {
            if let Some(next_token) = current_token.next_token() {
                current_token = next_token;
                break false;
            } 
        } else if let Some(token) = current_token.prev_token() {
            current_token = token;
        } else {
            break true;
        }
    }

```